thread-manager/src/lib.rs

```diff
@@ -0,0 +1,380 @@
+use {
+    anyhow::Ok,
+    serde::{Deserialize, Serialize},
+    std::{collections::HashMap, ops::Deref, sync::Arc},
+};
+
+pub mod native_thread_runtime;
+pub mod policy;
+pub mod rayon_runtime;
+pub mod tokio_runtime;
+
+pub use {
+    native_thread_runtime::{JoinHandle, NativeConfig, NativeThreadRuntime},
+    policy::CoreAllocation,
+    rayon_runtime::{RayonConfig, RayonRuntime},
+    tokio_runtime::{TokioConfig, TokioRuntime},
+};
+pub type ConstString = Box<str>;
+
+#[derive(Default, Debug)]
+pub struct ThreadManagerInner {
+    pub tokio_runtimes: HashMap<ConstString, TokioRuntime>,
+    pub tokio_runtime_mapping: HashMap<ConstString, ConstString>,
```

Comment:
is the idea here that we essentially have a mapping like so:
service --> tokio name --> tokio runtime
And service --> tokio name could many to 1 but the tokio name --> tokio runtime is 1 to 1?

Reply:
yes, exactly! I believe we will eventually have maybe 1-2 tokio runtimes + 2-3 rayons. and mappings to those.

Reply:
cool! I think it makes sense. Maybe we can get rid of the intermediate mapping and go just from service to runtime (I believe you mentioned this elsewhere)

Reply:
So, do you want the intermediate mapping removed now or can we keep it in?